### ISSUE TITLE
Thanks and future plans

### ISSUE THREAD

**Issue by matthew-a-dunlap** [shooking/ZoomPedalFun#6]
Thanks for diving into these pedals and uncovering info!

I have a Zoom G1XFour and am really excited to set its BPM/switch-presets/modulate-parameters on it via a hardware.

I have experience doing midi conversion with Arduino, and I picked up an extra usb host shield to hopefully get this working. My plan is to convert 5-pin CC messages into the sysex the G1Four needs.

Anyways, thanks again for doing this, and I'll share here how my explorations go.

**Comment by shooking** [comment]
Have you seen YT i made about Tap Tempo?
https://youtu.be/zwluLuraEJQ

I decide to use Pi rather than Arduino because USB midi just works. I use a 22" touchscreen to control. I should think about adding a touch tempo button and certainly add master.volume dial.a tempo in.

**Comment by matthew-a-dunlap** [comment]
I hadn't seen this! I saw a script where you have the sysex string for tempo, this also helps :)

Totally makes sense to use a Pi. I don't have a pi and am already really accustomed to the Arduino midi library so I'm going that route. But if the host board I got is too much a pain we'll see where I go next.

It'll be nice to set the bpm via my synth (I can have it send a cc, I don't need to worry about tap tempo).

I'm using this as an fx send for an electronic music setup. It works perfectly as my hardware has a stereo main out and one available mono channel as a send. I can plug both of those in and use the Zoom for delays and other weirdness, and its sums everything up.

**Comment by shooking** [comment]
Same here. I use the Zooms in FX loop of mixer to give new textures to old synths (like K1r). These beasts give our stereo. Not sure if the guitar input is mono but since you likely mix your tracks you got possibilities.

I also got some of the older Zoom pedals (midi ones).
The reason for Pi4 was all my home PCs are old. So unuse the 8Gb pi4 to video edit, control stuff etc. And of course a touchscreen can emulate a ton of controls.

I have various Arduino and ESP32 (good for bluetooth). I saw some issues with those USB shield (voltages, availability, cost!). There are links on the main readme to other projects where dudes have used Arduino etc. Sr Mousa comes to mind. Best of luck
